controllers/imagejob/imagejob_controller.go

```diff
@@ -365,11 +369,30 @@ func (r *Reconciler) handleNewJob(ctx context.Context, imageJob *eraserv1.ImageJ
 			return err
 		}
 		log.Info("Started "+containerName+" pod on node", "nodeName", nodeName)
+		namespacedNames = append(namespacedNames, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace})
+	}
+
+	for _, namespacedName := range namespacedNames {
+		if err := wait.PollImmediate(time.Nanosecond, time.Minute*5, r.isPodReady(namespacedName)); err != nil {
+			log.Error(err, "timed out waiting for pod to leave pending state", "pod NamespacedName", namespacedName)
+		}
 	}
 
 	return nil
 }
 
+func (r *Reconciler) isPodReady(namespacedName types.NamespacedName) wait.ConditionFunc {
+	return func() (bool, error) {
+		currentPod := &corev1.Pod{}
+
+		if err := r.Get(context.TODO(), namespacedName, currentPod); err != nil {
+			return false, client.IgnoreNotFound(err)
+		}
+
+		return currentPod.Status.Phase != corev1.PodPhase(corev1.PodPending) && currentPod.Status.Phase != corev1.PodPhase(corev1.PodUnknown), nil
```

Comment:
`PodUnknown` is deprecated, unused since 2015, so I think it's safe to ignore that case.

https://pkg.go.dev/k8s.io/api/core/v1#PodPhase